### PR TITLE
Add system-wide cache & cache to local_register

### PR DESCRIPTION
**Pull request by Baranovsky1986**
Add system-wide cache & cache to local_register

### CODE REVIEW

File: location_register/views/ratu_viewsets.py [hunk 1]
```diff
@@ -1,93 +1,39 @@
...
+from django.utils.decorators import method_decorator
 from location_register.models.ratu_models import Region, District, City, Citydistrict, Street
 from location_register.serializers.ratu_serializers import RegionSerializer, DistrictSerializer, CitySerializer, CitydistrictSerializer, \
     StreetSerializer
+from rest_framework import generics, viewsets
+from rest_framework.pagination import PageNumberPagination
 
 
-class RegionView(viewsets.ReadOnlyModelViewSet, PageNumberPagination):
+class CachedViewMixin:
```
Comment (criticus) [inline location_register/views/ratu_viewsets.py:10]: if this is an abstract class consider to move it to common file

File: location_register/views/ratu_viewsets.py [hunk 2]
```diff
@@ -1,93 +1,39 @@
...
+    @method_decorator(cache_page(60*15))
+    def dispatch(self, *args, **kwargs):
+        return super().dispatch(*args, **kwargs)
+    
+class RegionView(CachedViewMixin,viewsets.ReadOnlyModelViewSet):
```
Comment (roman-tiukh) [inline location_register/views/ratu_viewsets.py:15]:
```suggestion
class RegionView(CachedViewMixin, viewsets.ReadOnlyModelViewSet):
```

File: location_register/views/drv_viewsets.py [hunk 3]
```diff
@@ -1,24 +1,15 @@
...
+from django.utils.decorators import method_decorator
 from location_register.models.drv_models import DrvBuilding
 from location_register.serializers.drv_serializers import DrvBuildingSerializer
+from rest_framework import generics, viewsets
+from rest_framework.pagination import PageNumberPagination
+from .ratu_viewsets import CachedViewMixin
 
 
-class DrvBuildingViewSet(viewsets.ReadOnlyModelViewSet, PageNumberPagination):
+class DrvBuildingViewSet(CachedViewMixin,viewsets.ReadOnlyModelViewSet):
```
Comment (roman-tiukh) [inline location_register/views/drv_viewsets.py:10]:
```suggestion
class DrvBuildingViewSet(CachedViewMixin, viewsets.ReadOnlyModelViewSet):
```

File: location_register/views/ratu_viewsets.py (hunk 1 above)
Comment (roman-tiukh) [reply location_register/views/ratu_viewsets.py:10]: +1
move it to common file

Comment (Baranovsky1986) [reply location_register/views/ratu_viewsets.py]: i did it

File: location_register/views/ratu_viewsets.py (hunk 2 above)
Comment (Baranovsky1986) [reply location_register/views/ratu_viewsets.py]: i did it

File: location_register/views/drv_viewsets.py (hunk 3 above)
Comment (Baranovsky1986) [reply location_register/views/drv_viewsets.py]: did it

File: data_converter/settings_local.base.py [hunk 4]
```diff
@@ -50,3 +52,17 @@
...
+        'BACKEND': 'django_redis.cache.RedisCache',
+        'LOCATION': 'redis://127.0.0.1:6379/',
+        'OPTIONS': {
+            "IGNORE_EXCEPTIONS": True,
+            # 'PASSWORD': 'XXXXXXXXX',
+            'CLIENT_CLASS': 'django_redis.client.DefaultClient',
+            
+        }
+    }
+}
```
Comment (roman-tiukh) [inline data_converter/settings_local.base.py:67]: add pls here DummyCache for example, and DummyCache will be default cache backend

Comment (Baranovsky1986) [reply data_converter/settings_local.base.py:67]: instead of redis?

File: data_converter/settings_local.base.py [hunk 5]
```diff
@@ -50,3 +52,17 @@
...
+
+CACHES = {
+    'default': {
+        'BACKEND': 'django_redis.cache.RedisCache',
+        'LOCATION': 'redis://127.0.0.1:6379/',
+        'OPTIONS': {
+            "IGNORE_EXCEPTIONS": True,
+            # 'PASSWORD': 'XXXXXXXXX',
+            'CLIENT_CLASS': 'django_redis.client.DefaultClient',
+
```
Comment (roman-tiukh) [inline data_converter/settings_local.base.py:64]:
```suggestion
         # 'BACKEND': 'django.core.cache.backends.dummy.DummyCache',  # use this cashe for developing
```

File: data_converter/settings_local.base.py (hunk 4 above)
Comment (roman-tiukh) [reply data_converter/settings_local.base.py:67]: add extra, check next comment

**Pull request merged**